### PR TITLE
[v0.26.0] Add Nhost to Auth Docs

### PR DESCRIPTION
**Pull request by nunopato**
- Adds Nhost to Auth docs

### CODE REVIEW

File: docs/authentication.md [hunk 1]
```diff
@@ -505,6 +506,41 @@ To complete setup, you'll also need to update your `api` server manually. See ht
...
+
+The following CLI command will install required packages and generate boilerplate code and files for Redwood Projects:
+
+```terminal
+yarn rw setup auth nhost
+```
+
+#### Setup
+
+Update your .env file with the following settings which can be found on your project's dashboard.
```
Comment (Tobbe) [inline docs/authentication.md:523]:
```suggestion
Update your .env file with the following setting which can be found on your Nhost project's dashboard.
```

**Review by Tobbe — APPROVED** [review]

**Comment by Tobbe** [comment]
This looks great @nunopato. Thanks for writing the docs.
We'll merge this when the auth provider is included in a release

**Comment by Tobbe** [comment]
Actually just remembered one thing. Did you want to mention `skipFetchCurrentUser` and tell people to add that? And also why they should add it?

**Comment by nunopato** [comment]
@Tobbe I am not sure, to be honest. The query for `getCurrentUser`, found [here](https://github.com/redwoodjs/redwood/blob/main/packages/auth/src/AuthProvider.tsx#L134), as expected, was failing for me because I pointed `apiProxyPath` to a Hasura endpoint on Nhost after removing the `./api` folder. It is not related to the auth provider, but at the same time, I don't understand why would the user be fetched from `apiProxyPath` when using any of the auth providers. I might be missing something here.

**Comment by Tobbe** [comment]
> I don't understand why would the user be fetched from `apiProxyPath` when using any of the auth providers.

We do this so our users can augment the info they get from their auth provider with info from their own database. Like if all you get from your auth provider is an email, maybe you have a table in your own db with name and roles connected to that email. `getCurrentUser` on the API side lets you fetch that from the db, and return that to the frontend. Here's an example from the code that gets generated when you run the auth setup

```javascript
export const getCurrentUser = async ({ email }) => {
  return await db.user.findUnique({ where: { email } })
}
```

**Comment by nunopato** [comment]
@Tobbe thank you for the explanation, then I think we are good to go since that is not specific to Nhost used as an auth provider.

**Comment by Tobbe** [comment]
> @Tobbe thank you for the explanation, then I think we are good to go since that is not specific to Nhost used as an auth provider.

So, if you use Nhost auth together with everything else on Nhost (hasura etc), you probably want `skipFetchCurrentUser`. But if, for some reason, you're only using Nhost for auth, and have everything else setup "the Redwood way" with apollo server and prisma you probably do want to call `getCurrentUser`.

I think a note about this in the docs would be great 👍

**Comment by nunopato** [comment]
@Tobbe thank you. Added the missing bit.

File: docs/authentication.md [hunk 2]
```diff
@@ -505,6 +506,52 @@ To complete setup, you'll also need to update your `api` server manually. See ht
...
+
+For the docs on Authentication, see: <https://docs.nhost.io/auth>
+
+If you are also using Nhost as your GraphQL API server, you will need to set `skipFetchCurrentUser` when creating the Nhost client, as follows:
+
+```js
+const nhostClient = createClient({
+  baseURL: 'https://backend-REPLACE.nhost.app',
+  autoLogin: false,
+  skipFetchCurrentUser: true,
```
Comment (Tobbe) [inline docs/authentication.md:548]: Sorry if I was unclear, but this is actually not how that setting is used. Instead you're supposed to add it as a prop to `<AuthProvider>`

Like this

```tsx
<AuthProvider client={nhost} type="nhost" skipFetchCurrentUser>
```

Comment (nunopato) [reply docs/authentication.md:548]: @Tobbe I am sorry, this is what happens when you do things late at night. What a miss 🙄

Comment (Tobbe) [reply docs/authentication.md]: Looking good now 👍

**Review by Tobbe — APPROVED** [review]

**Comment by Tobbe** [comment]
As soon as https://github.com/redwoodjs/redwood/pull/1725 is part of a RW release we can merge this

**Review by thedavidprice — APPROVED** [review]
Thanks! I have this on the list to merge with v0.26.0 release.

**Pull request merged**